xarray/core/dataarray.py

```diff
@@ -6718,7 +6716,7 @@ def groupby(
 
         Parameters
         ----------
-        group : Hashable or DataArray or IndexVariable or mapping of Hashable to Grouper
+        group : str or DataArray or IndexVariable or iterable of Hashable or mapping of Hashable to Grouper
```

Comment:
should this be a sequence instead of iterable? In other words, should this accept something like `dict` or `set` (since it's possible to iterate over `set`/`dict` and get hashables)?